com.unity.netcode.gameobjects/Components/NetworkAnimator.cs

```diff
@@ -392,28 +391,15 @@ internal struct AnimationMessage : INetworkSerializable
             // Not to be serialized, used for processing the animation message
             internal bool HasBeenProcessed;
 
-            // state hash per layer.  if non-zero, then Play() this animation, skipping transitions
             internal List<AnimationState> AnimationStates;
 
-            /// <summary>
-            /// Resets all AnimationStates' IsDirty flag
-            /// </summary>
-            internal void ClearDirty()
-            {
-                if (AnimationStates == null)
-                {
-                    return;
-                }
-                for (int i = 0; i < AnimationStates.Count; i++)
-                {
-                    var animationState = AnimationStates[i];
-                    animationState.IsDirty = false;
-                    AnimationStates[i] = animationState;
-                }
-            }
+            // Determines how many AnimationState entries are dirty
+            // this is a 1:1 mapping of the AnimationStates indices
```

Comment:
This comment is not as clear as it could be.
If it is a count, how can it also be a mapping what does it map to what?

Reply:
The List<AnimationState> AnimationStates indices values... 
Example:
If you have an IsDirtyCount of 5 then AnimationStates[0] through AnimationStates[4] would contain the states to be updated.  I will see if I can improve that comment.  When initializing the AnimationStates list and synchronizing a newly joined client, I use the layers count and wanted to make sure that it was clear that AnimationStates is not a 1:1 mapping of layers but is used to hold the AnimationState for each layer that had changed.